src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentStepExecutionCallback.java

```diff
@@ -0,0 +1,101 @@
+package com.atlassian.bitbucket.jenkins.internal.deployments;
+
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeploymentEnvironment;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.DeploymentState;
+import com.atlassian.bitbucket.jenkins.internal.scm.BitbucketSCMRepository;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import com.google.common.annotations.VisibleForTesting;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.steps.BodyExecutionCallback;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+
+import java.util.logging.Logger;
+
+import static java.lang.String.format;
+
+@VisibleForTesting
+public class DeploymentStepExecutionCallback extends BodyExecutionCallback {
+
+    private static final Logger LOGGER = Logger.getLogger(DeploymentStepExecutionCallback.class.getName());
+    private static final long serialVersionUID = 1L;
```

Comment:
What is this long used for?

Reply:
In general, Java saves `serialVersionUID` during serialization and checks that it matches during deserialization. If `serialVersionUID` is not defined in the class, then the serializer computes something, but that may make the data more difficult to deserialize later if more fields are added to the class.

Jenkins saves the states of in-progress pipelines to files and I believe this callback can be serialized at that time.